### PR TITLE
Score tests

### PR DESCRIPTION
**Pull request by dirkmc**
Part of https://github.com/libp2p/go-libp2p-pubsub/issues/287
- [x] Time in mesh
- [x] First message deliveries
- [x] Mesh message deliveries
- [x] Mesh failure penalty
- [x] Invalid message deliveries
- [x] App specific score
- [x] IP colocation factor

### CODE REVIEW

File: score_test.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+	params := &PeerScoreParams{
+		AppSpecificScore: func(peer.ID) float64 { return 0 },
+		Topics:           make(map[string]*TopicScoreParams),
+	}
+	topicScoreParams := &TopicScoreParams{
+		TopicWeight:       1,
+		FirstMessageDeliveriesWeight:  1,
+		FirstMessageDeliveriesDecay: 1.0, // test without decay for now
+		FirstMessageDeliveriesCap: 2000,
+		TimeInMeshQuantum: time.Second, // bug? not setting this causes a div by zero
```
Comment (yusefnapora) [inline score_test.go]: @vzyo - I hit a panic when this is unset, since we divide by it when calculating the time in mesh score.

Comment (vyzo) [reply score_test.go]: ouch -- we should fix the validation to catch this and return an error.
once we do that let's also update the validation test to keep the score_params coverage to 100% :)

**Comment by yusefnapora** [comment]
reading through, I just realized I'm using the `withinVariance` helper in places where I don't really need to... we should be able to do exact comparisons for the scores that aren't based on time.

**Comment by vyzo** [comment]
good point.

**Comment by vyzo** [comment]
failure in `TimeInMesh`:
```
=== RUN   TestScoreTimeInMesh

    TestScoreTimeInMesh: score_test.go:47: Score: 35.000000. Expected 20.000000 ± 10.000000

```

**Review by vyzo — APPROVED** [review]
looks great!

**Pull request merged**